qendpoint-store/src/main/java/com/the_qa_company/qendpoint/store/EndpointStore.java

```diff
@@ -115,9 +115,9 @@ public class EndpointStore extends AbstractNotifyingSail {
 	File checkFile;
 
 	// flag if the store is merging or not
-	private boolean isMerging = false;
+	private volatile boolean isMerging = false;
 
-	public boolean isMergeTriggered = false;
+	public volatile boolean isMergeTriggered = false;
```

Comment:
These needed to be volatile for me to read updated values from a different thread.